t/01-unit/cache_key.t

```diff
@@ -330,14 +330,17 @@ location /t {
         local vary_key = generate_vary_key({"Foo", "X-Test"}, callback, nil)
         log(vary_key)
         assert(called_flag == true, "Callback is called - multivalue spec")
-        assert(vary_key == "foo:bar:x-test:value", "Vary spec not modified with noop function - multivalue spec")
+        assert(string.find(vary_key, "foo:bar"), "Vary spec not modified with noop function - multivalue spec")
```

Comment:
This wouldn't catch an addition to the `vary_key`, might be worth just checking the overall length of the string too?

Reply:
Thanks, I've added that now.

Reply:
Would these sort of checks be better done as something like this:
```
        assert(string.find(vary_key, "foo:bar"), "Vary spec not modified with noop function - multivalue spec checking for foo-bar")
        vary_key = string.gsub(vary_key, "foo:bar", "")
        assert(string.find(vary_key, "x-test:value"), "Vary spec not modified with noop function - multivalue spec checking for x-test")
        vary_key = string.gsub(vary_key, "x?-test:value", "")
        assert(vary_key == ":", "Vary spec not modified with noop function - multivalue spec looking for :")
```

Reply:
There's a risk of being 'too clever' in these tests though, keeping it as simple as possible means any bizarre unexpected values would get caught.
You could parse the string, split it by `:` and check that all the expected values are there and no unexpected values are. 
🤷‍♂️  probably fine as it is now